javascripts/tagmanager.js

```diff
@@ -742,6 +742,17 @@
                     if (!node) {
                         return;
                     }
+
+                    // If the content belongs to a masked element and it doesn't have any children, return a masked string.
+                    if (TagManager.dom.shouldElementBeMasked(node) && node.children.length === 0) {
+                        return '*******';
+                    }
+
+                    // If the element has children that should be masked, deal with that.
+                    if (TagManager.dom.elementHasMaskedChild(node)) {
```

Comment:
@snake14 Why do we need this ? Above code will do the needful correct ?

Reply:
@AltamashShaikh After looking more at the example you provided and the code in the heatmap plugin, I ended up handling masking part of the content of an element instead of everything. This is especially important if a div is masked, but one of its children has the unmask attribute. So, lines 747-749 handle masking elements that should be masked and don't have any children. The if statement on line 752 checks if the element has children elements that need to be masked.

Reply:
@snake14 Understood now, we need this change